File: local-claude-ollama/examples/javascript/streaming_chat.mjs
/**
 * Streaming chat example with Ollama backend.
 *
 * Prerequisites:
 *   npm install @anthropic-ai/sdk
 *
 * Usage:
 *   node streaming_chat.mjs
 */

import Anthropic from "@anthropic-ai/sdk";

const client = new Anthropic({
  baseURL: "http://localhost:11434",
  apiKey: "ollama",
});

async function main() {
  console.log("Streaming response:");
  console.log("-".repeat(40));

  const stream = client.messages.stream({
    model: "qwen3-coder",
    max_tokens: 1024,
    messages: [
      {
        role: "user",
        content: "Explain async/await in JavaScript with examples.",
      },
    ],
  });

  // Process stream events
  for await (const event of stream) {
    if (
      event.type === "content_block_delta" &&
      event.delta.type === "text_delta"
    ) {
      process.stdout.write(event.delta.text);
    }
  }

  console.log();
  console.log("-".repeat(40));

  // Get final message
  const finalMessage = await stream.finalMessage();
  console.log(`Total tokens: ${finalMessage.usage.input_tokens + finalMessage.usage.output_tokens}`);
}

main().catch(console.error);

File: local-claude-ollama/examples/python/streaming_chat.py
#!/usr/bin/env python3
"""
Streaming chat example with Ollama backend.

Prerequisites:
    pip install anthropic

Usage:
    python streaming_chat.py
"""

import anthropic


def main():
    client = anthropic.Anthropic(
        base_url="http://localhost:11434",
        api_key="ollama",
    )

    print("Streaming response:")
    print("-" * 40)

    # Use streaming for real-time output
    with client.messages.stream(
        model="qwen3-coder",
        max_tokens=1024,
        messages=[
            {
                "role": "user",
                "content": "Explain the concept of recursion with a simple example.",
            }
        ],
    ) as stream:
        for text in stream.text_stream:
            print(text, end="", flush=True)

    print()
    print("-" * 40)


if __name__ == "__main__":
    main()
